lumicks/pylake/kymo.py

```diff
@@ -104,14 +104,13 @@ def _plot(self, image, **kwargs):
         import matplotlib.pyplot as plt
 
         size_um = self.size_um[0]
-        ts = self.timestamps
-        duration = (ts[0, -1] - ts[0, 0]) / 1e9
-        linetime = (ts[0, 1] - ts[0, 0]) / 1e9
+        duration = self.line_time_seconds * image.shape[1]
+        linetime = self.line_time_seconds
 
         default_kwargs = dict(
             # With origin set to upper (default) bounds should be given as (0, n, n, 0)
             # pixel center aligned with mean time per line
-            extent=[-0.5 * linetime, duration + 0.5 * linetime, size_um, 0],
+            extent=[-0.5 * linetime, duration - 0.5 * linetime, size_um, 0],
```

Comment:
In the old one, the timestamps corresponded to the center of the first pixel (`0`) and last (`size - 1`). I would argue that before, the duration was actually a misnomer, since it was the duration from the first center to the last center rather than the actual duration of the kymo (which is extent to extent).

In the new setup, for a pixel array with size 4:
`0 | 1 | 2 | 3`

You can see how the extent is `-0.5 dx` to `3 + 0.5 * dx` (or `4 - 0.5 * dx`, or `size - 0.5 * dx`).